File: 1813-maximum-erasure-value/maximum-erasure-value.java
class Solution {
    public int maximumUniqueSubarray(int[] nums) {
        HashSet<Integer> set = new HashSet();
        int sum = 0;
        int maxSum = 0;
        int start=0;
        int end=0;
        while(end<nums.length){
            if(!set.contains(nums[end])){
                sum+=nums[end];
                set.add(nums[end]);
                end++;
            }
            else{
                maxSum = Math.max(maxSum,sum);
                while(start<end&&set.contains(nums[end])){
                    sum-=nums[start];
                    set.remove(nums[start]);
                    start++;
                }
            }
        }
                maxSum = Math.max(maxSum,sum);

        return maxSum;

    }
}

File: 1813-maximum-erasure-value/maximum-erasure-value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        int maxsum=0;
        int sum=0;
        unordered_set<int> ans;
        int end=0;
        int start=0;
        while(end<nums.size()){
            if(ans.find(nums[end]) == ans.end()){
                ans.insert(nums[end]);
                sum=sum+nums[end];
                maxsum=max(maxsum,sum);
                end++;
            }
            else{
                    ans.erase(nums[start]);
                    sum=sum-nums[start];
                    start=start+1;
             }
            
        }
        return maxsum;
    }
};